tests/foreman/api/test_repository.py

```diff
@@ -208,23 +221,26 @@ def test_update(self, attrs):
         @Feature: Repository
 
         """
-        client.put(
-            self.repository.path(),
-            attrs,
-            auth=get_server_credentials(),
-            verify=False,
-        ).raise_for_status()
-        real_attrs = self.repository.read_json()
-        for name, value in attrs.items():
-            self.assertIn(name, real_attrs.keys())
-            if name == 'content_type':
-                # Cannot update a repository's content type.
-                self.assertEqual(
-                    self.repository.get_fields()['content_type'].default,
-                    real_attrs[name]
-                )
-            else:
-                self.assertEqual(value, real_attrs[name])
+        for attrs in _test_data():
+            with self.subTest(attrs):
+                client.put(
```

Comment:
Is there a reason why we're using `client.put`? Is this 'legacy' code back from when maybe one could not create or update a repository?